csm/core/services/security.py

```diff
@@ -350,3 +350,37 @@ async def check_certificate_expiry_time_task(self):
                            tzinfo=timezone.utc),
             interval=timedelta(days=1)
         )
+
+    async def _get_name_details(self, name):
+        """
+        Get x509 Name object details (i.e Subject and Issuer)
+        """
+        name_details = {}
+        for k,v in x509.oid._OID_NAMES.items():
+            try:
+                name_details[v] = name.get_attributes_for_oid(k)[0].value
+            except Exception as e:
+                Log.error(f"No value for {v}:{e}")
+        return name_details
+
+    async def get_certificate_details(self):
+        """
+        Get current certificate details
+        """
+        path = Conf.get(const.CSM_GLOBAL_INDEX, "HTTPS.certificate_path")
```

Comment:
Make sure that the code is inline with Conf Store.

Reply:
ConfStore PR is not yet merged. So it can be taken care once that PR is merged.

Reply:
I would say make the changes in this PR itself, so that no re-work is required. Once Prathamesh's PR will be merged your code will work. Since, currently CSM UI is only consumer as of now and the work is not started yet.
If you want to make the changes afterwards, so may be create a ticket for it so that you remember to updated the code,
@kupranay  Please suggest.

Reply:
Updated the code to align with ConfStore